### PR TITLE
Fix problem introduced by clang-format include reordering

### PR DESCRIPTION
**Pull request by lrusak**
While working on some other stuff I came across an issue due to clang-format reordering.

This issue shows up after the first commit in this PR and is fixed by the second commit (I would like to squash them before merging). I don't think this is the best solution though and the defines should probably be namespaced to avoid polluting the global namespace. I feel like this was just working by chance before but now we might run into issues like this with clang-format.

```
In file included from /home/lukas/Documents/git/xbmc/xbmc/interfaces/legacy/Dialog.cpp:9:
/home/lukas/Documents/git/xbmc/xbmc/interfaces/legacy/Dialog.h:24:31: error: expected identifier before numeric constant
   24 | #define INPUT_TIME            3
      |                               ^
/home/lukas/Documents/git/xbmc/xbmc/dialogs/GUIDialogNumeric.h:28:21: note: in expansion of macro ‘INPUT_TIME’
   28 |   enum INPUT_MODE { INPUT_TIME = 1, INPUT_DATE, INPUT_IP_ADDRESS, INPUT_PASSWORD, INPUT_NUMBER, INPUT_TIME_SECONDS };
      |                     ^~~~~~~~~~
```

Suggestions?

### CODE REVIEW

**Comment by fritsch** [comment]
What about making the enum a class and use it as such with Class::Value ?

**Comment by lrusak** [comment]
> What about making the enum a class and use it as such with Class::Value ?
the enum is already guarded by the class https://github.com/xbmc/xbmc/blob/master/xbmc/dialogs/GUIDialogNumeric.h#L24-L28

I think the defines are being declared first by the preprocessor and ideally should be namespaced.

**Comment by fritsch** [comment]
When I read the code I was totally not sure when which one is used. Also they have different values. That's why I would clearly separate them by having same semantic value, but different types. So that it is entirely clear when using which one.

File: xbmc/interfaces/legacy/Dialog.h [hunk 1]
```diff
@@ -18,15 +18,15 @@
...
-#define ALPHANUM_HIDE_INPUT   2
+const int INPUT_ALPHANUM{0};
+const int INPUT_NUMERIC{1};
+const int INPUT_DATE{2};
+const int INPUT_TIME{3};
+const int INPUT_IPADDRESS{4};
+const int INPUT_PASSWORD{5};
+
+const int PASSWORD_VERIFY{1};
+const int ALPHANUM_HIDE_INPUT{2};
```
Comment (yol) [inline xbmc/interfaces/legacy/Dialog.h]: Optimal solution: Make this `enum class` and put into namespace. But just putting it into a namespace should be enough for now. Also I think you can make them `constexpr`

**Comment by Paxxi** [comment]
there's a couple of these including some "unfixable" ones that I found when I ran clang-format on the whole repo, nice to get some of these fixed

**Comment by lrusak** [comment]
So is this an acceptable solution then?

**Comment by Paxxi** [comment]
Looks good to me. For a proper fix we shouldn't have those enum values duplicated but that's a larger refactoring and probably involves a ton of work. This is a good fix for the time being.

**Comment by lrusak** [comment]
dammit, I forgot to squash and merge :unamused:

**Comment by Paxxi** [comment]
We'll live 😁

**Pull request merged**